src/main/java/com/gmail/goosius/siegewar/playeractions/PlaceBlock.java

```diff
@@ -315,5 +310,13 @@ private static int getNumActiveAttackSieges(Nation nation) {
 		}
 		return result;
 	}
+
+	private static boolean isValidPlacement(Block block) {
+		if (!TownyAPI.getInstance().isWilderness(block) && isWhiteBanner(block))
+			return true;
+		if (SiegeWarBlockUtil.getCardinalAdjacentTownBlocks(block).size() > 0)
+			return true;
```

Comment:
:construction: 
- With the current code in this PR, IF someone places a coloured banner inside a town, the execution flow will get to the second line above and (pointlessly) search for the cardinal town blocks.
- Please revise the logic

Reply:
Fixed 👍